src/connectors/menu/connectMenu.js

```diff
@@ -164,7 +164,7 @@ export default function connectMenu(renderFn, unmountFn = noop) {
           showMore ? showMoreLimit : limit
         );
 
-        return widgetConfiguration;
+        return configuration.setQueryParameters(widgetConfiguration);
```

Comment:
Since now we use a `SearchParameters` we can use the function to add the facets and avoid the call to the merge function inside `enhanceConfiguration`. It means that `getConfiguration` should not override the "controlled" parameters e.g `hierarchicalFacets`.

Reply:
My bad I thought `SearchParameters` would do it for us https://github.com/algolia/instantsearch.js/pull/3976#discussion_r308155858.